libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKAuthPreferences.m

```diff
@@ -93,13 +87,13 @@ - (NSString *)loginHost {
         [defaults synchronize];
         return host;
     }
-    
-    // Fetch from the standard defaults or bundle.
+
+    // Fetch from the standard defaults or bundle and ensues that login host still exists.
     NSString *loginHost = [defaults stringForKey:kSFUserAccountOAuthLoginHost];
-    if ([loginHost length] > 0) {
+    if (loginHost.length > 0 && [[SFSDKLoginHostStorage sharedInstance] loginHostForHostAddress:loginHost] != nil) {
```

Comment:
This is the other important fix. Here, we check if `NSUserDefaults` has a value stored that's not persisted in our server list, i.e. a zombied server host. If so, we replace it with the new value (this is handled in the code below).